### PR TITLE
Updates and Enhancements to AOMP Recipe. Added version 3.10.0.

### PR DESCRIPTION
**Pull request by estewart08**
Reduced repitition with version hashes.
Expanded dependency versioning.
Reduced repitition with cmake args.
Added version 3.10.0.

### CODE REVIEW

**Comment by estewart08** [comment]
@adamjstewart Similar updates/enhancements applied from rocm-openmp-extras. Thanks for all the reviews.

File: var/spack/repos/builtin/packages/aomp/package.py [hunk 1]
```diff
@@ -4,104 +4,121 @@
...
-    version('3.5.0', sha256=aomp35['aomp'])
+    version('3.10.0', sha256=versions_dict['3.10.0']['aomp'])
+    version('3.9.0', sha256=versions_dict['3.9.0']['aomp'])
+    version('3.8.0', sha256=versions_dict['3.8.0']['aomp'])
+    version('3.7.0', sha256=versions_dict['3.7.0']['aomp'])
+    version('3.5.0', sha256=versions_dict['3.5.0']['aomp'])
+
+    # Cmake above 3.18 would fail the build on 3.5.0
+    depends_on('cmake@3:', type='build')
+    depends_on('cmake@3:3.17.4', when='@3.5.0', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/aomp/package.py]:
```suggestion
    depends_on('cmake@3:3.17', when='@3.5.0', type='build')
```

File: var/spack/repos/builtin/packages/aomp/package.py [hunk 2]
```diff
@@ -4,104 +4,121 @@
...
+
+    # Python 2 is needed for 3.5.0 and 3.8.0, limit py-setuptools
+    # to avoid spec error
+    depends_on('python@2.7.8:2.7.18', when='@3.5.0:3.8.0', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/aomp/package.py]: Python 2.7.7 doesn't work?
```suggestion
    depends_on('python@2.7:2.8', when='@3.5.0:3.8.0', type='build')
```

File: var/spack/repos/builtin/packages/aomp/package.py [hunk 3]
```diff
@@ -4,104 +4,121 @@
...
+    depends_on('py-setuptools@44.1.0:44.99.99', when='@3.5.0:3.8.0',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/aomp/package.py]:
```suggestion
    depends_on('py-setuptools@:44', when='@3.5.0:3.8.0',
```

File: var/spack/repos/builtin/packages/aomp/package.py [hunk 4]
```diff
@@ -507,31 +412,22 @@ def install(self, spec, prefix):
...
             '-DOPENMP_ENABLE_LIBOMPTARGET_HSA=1'
+            '-DCMAKE_CXX_FLAGS=-g',
+            '-DCMAKE_C_FLAGS=-g'
         ]
 
         if self.spec.version >= Version('3.9.0'):
             components['openmp-debug'] += [
                 '-DENABLE_SOURCE_COPY=ON',
-                '-DOPENMP_SOURCE_DEBUG_MAP={0}'.format(debug_map_format),
+                '-DOPENMP_SOURCE_DEBUG_MAP="{0}"'.format(debug_map_format)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/aomp/package.py]: I don't think quotes are needed here.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/aomp/package.py (hunk 1 above)
Comment (estewart08) [reply var/spack/repos/builtin/packages/aomp/package.py]: done

File: var/spack/repos/builtin/packages/aomp/package.py (hunk 2 above)
Comment (estewart08) [reply var/spack/repos/builtin/packages/aomp/package.py]: done

File: var/spack/repos/builtin/packages/aomp/package.py (hunk 3 above)
Comment (estewart08) [reply var/spack/repos/builtin/packages/aomp/package.py]: done

File: var/spack/repos/builtin/packages/aomp/package.py (hunk 4 above)
Comment (estewart08) [reply var/spack/repos/builtin/packages/aomp/package.py]: done

File: var/spack/repos/builtin/packages/aomp/package.py [hunk 5]
```diff
@@ -4,104 +4,121 @@
...
+    version('3.5.0', sha256=versions_dict['3.5.0']['aomp'])
+
+    # Cmake above 3.18 would fail the build on 3.5.0
+    depends_on('cmake@3:', type='build')
+    depends_on('cmake@3:3.17', when='@3.5.0', type='build')
+
+    # Python 2 is needed for 3.5.0 and 3.8.0, limit py-setuptools
+    # to avoid spec error
+    depends_on('python@2.7:2.8', when='@3.5.0:3.8.0', type='build')
+    depends_on('py-setuptools@44', when='@3.5.0:3.8.0',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/aomp/package.py]:
```suggestion
    depends_on('py-setuptools@:44', when='@3.5.0:3.8.0',
```

File: var/spack/repos/builtin/packages/aomp/package.py [hunk 6]
```diff
@@ -4,104 +4,121 @@
...
+    depends_on('py-setuptools', when='@3.9.0:', type='build')
 
-    depends_on('cmake@3.5.2:3.13.4', type='build')
     depends_on('mesa18~llvm@18.3:', type=('build', 'link'))
-    depends_on('py-setuptools@44.1.0', type='build')
-    depends_on('python@2.7.18', type='build', when='@3.5.0:3.8.0')
-    depends_on('python@3.6.9', type='build', when='@3.9.0:')
     depends_on('py-pip', when='@3.8.0:', type='build')
-    depends_on('py-wheel@0.29.0', when='@3.8.0:', type=('build', 'run'))
+    depends_on('py-wheel', when='@3.8.0:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/aomp/package.py:115]: Not sure if I've asked this in the past, but are pip and wheel actually needed?

Comment (estewart08) [reply var/spack/repos/builtin/packages/aomp/package.py:115]: Yes, they are needed during the ompd portion of the openmp build.

**Review by adamjstewart — APPROVED** [review]

**Comment by estewart08** [comment]
This all set to merge now?

**Pull request merged**